feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.ts

```diff
@@ -34,18 +34,20 @@ export class ConfiguratorAttributeCheckBoxListComponent
   ngOnInit(): void {
     const disabled = !this.allowZeroValueQuantity;
 
-    for (const value of this.attribute.values) {
-      let attributeCheckBoxForm;
-
-      if (value.selected === true) {
-        attributeCheckBoxForm = new FormControl({
-          value: true,
-          disabled: disabled,
-        });
-      } else {
-        attributeCheckBoxForm = new FormControl(false);
+    if (this.attribute.values) {
+      for (const value of this.attribute.values) {
```

Comment:
why we cannot write `for (const value of this.attribute?.values)` with a question mark and do not use if-condition

Reply:
Because values is optional, and thus can be undefined